src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -97,6 +97,10 @@ public function load(array $configs, ContainerBuilder $container)
             if (is_dir($dir = $bundle['path'].'/Resources/views')) {
                 $this->addTwigPath($twigFilesystemLoaderDefinition, $dir, $name);
             }
+
+            if (null !== $bundle['parent']) {
+                $this->prependParentPaths($twigFilesystemLoaderDefinition, $container, $name, $bundle['parent']);
```

Comment:
`prependParentPaths` was confusing at first. Maybe keep actual path registration explicit in `load()`

```php
while ($parent) {
   // add parent
}

// add self
```

Basically we dont need any path prepending or recursion, only adding paths with the right calling order.